### PR TITLE
Joss paper

### PR DESCRIPTION
**Pull request by mshamrai**
Feel free to comment and suggest improvements

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+title: 'LangToGroup: A Haskell package for group presentation construction from formal language'
+tags:
+  - Haskell
+  - formal languages
+  - group theory
+  - Turing machines
+  - group presentation construction
+authors:
+  - name: Semyon Grigorev
```
Comment (gsvgit) [inline paper/paper.md:10]: I'm not sure that I should be the first. Project lead traditionally last in the list of authors.

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Inspired by this, we wondered what if we treat formal language as a group presentation? This will allow us to use various group theoretical methods for formal languages. Moreover, it allows experimenting with computational group theory algorithms [@GAP4]. But construction by hand like that takes a lot of time and paper. So it requires an efficient tool for automatic group presentation construction from formal language.
+
+# Statement of need
+
+LangToGroup is a Haskell console application for group presentation construction. Haskell was chosen due to its rich and convenient type system,
```
Comment (gsvgit) [inline paper/paper.md:43]: It is not a statement of need. It is a solution description.

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+which we apply to represent algorithmic types such as formal grammar,
+Turing machine, S-machine, or group presentation. Furthermore, it is worth noting that data types were intentionally described in the same way as in the definitions of the corresponding concepts. This was done to maintain the maximum similarity to the article [@Sapir]. 
+
+The package provides the capability to obtain a presentation of a group from a Turing machine in two ways described in [@Sapir] and [@rotman1995word]. The algorithms to construct a Turing machine from context-free languages and Boolean languages have developed as well. Moreover, LangToGroup supports several types of input and output and each step of the construction can be printed in a document of LaTeX format for convenient usage. Therefore, LangToGroup presents a full pipeline of group presentation construction from formal languages and also can be used for construction from Turing machines.
```
Comment (gsvgit) [inline paper/paper.md:47]: This paragraph is also a solution description.

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Mainly LangToGroup package has been developed for research in disciplines such as official languages, computational theory, and group theory, but this may be of interest to researchers in many other fields. The tool connects "computation" with group theory. That allows us to think mathematically about computational processes and use the concepts of group theory to analyze and solve problems.
```
Comment (gsvgit) [inline paper/paper.md:49]: This one looks like a statement of need. More or less.

**Review by gsvgit — CHANGES_REQUESTED** [review]
The structure of the paper should be improved. 
1) Statement of need should be added.
2) Current "Statement of need" section should be renamed to "Solution description"
3) I guess, more discussion on the tool functionality should be added.

**Comment by MaxVortman** [comment]
GitHub action for paper compilation to pdf format is added. Current version can be downloaded from "Artifacts" section at https://github.com/MaxVortman/LangToGroup/actions/runs/1447886259

**Review by gsvgit — APPROVED** [review]

**Pull request merged**